addons/website_sale/controllers/main.py

```diff
@@ -152,12 +153,20 @@ def _get_pricelist_context(self):
 
         return pricelist_context, pricelist
 
-    def _get_compute_currency(self, pricelist, product=None):
-        company = product and product._get_current_company(pricelist=pricelist, website=request.website) or pricelist.company_id or request.website.company_id
-        from_currency = (product or request.env['res.company']._get_main_company()).currency_id
-        to_currency = pricelist.currency_id
+    def _get_compute_currency_pricelist_to_pricelist(self, pricelistFrom, pricelistTo, product=None):
+        company = product and product._get_current_company(pricelist=pricelistTo, website=request.website) or pricelistTo.company_id or request.website.company_id
```

Comment:
`pricelistTo` is passed as `pricelist` argument but it is not a pricelist all the time?
If coming from `_get_compute_currency_pricelist_to_product()`, pricelistTo is a product or a company record, won't that crash or at least have unexpected results?

Apart from that, it is really hard to understand those methods given the method name and the parameters name (pricelist for a product or a company eg)
Could you add a small docstring to those methods, that would definitely help anyone having to go through it :yum: